### PR TITLE
Spring Cloud AWS RDS

### PR DESCRIPTION
**Pull request by pratikdas**
Please re-read your article a day or so after you finished writing it and work through this checklist to polish the quality. This will save all of us a lot of time during review.

## Content
- [ ] the article explains *why* and not only *how*
- [ ] (half-)sentences that express a meaningful idea that is understandable by itself own are **bold** to allow quick scanning of the text

## Text
- [ ] the text is conversational ("we" instead of "you", easy language)
- [ ] there are no "walls of text" (short sentences, short paragraphs with a single idea)
- [ ] the text is inclusive ("the developers" and "they" instead of "the developer" and "he/she")
- [ ] the text uses active voice instead of passive voice
- [ ] names are spelled consistently throughout the article (correct uppercase / lowercase)
- [ ] file names and variables are highlighted as `code`

## Structure
- [ ] the article has a short intro before the first headline
- [ ] by reading the table of contents (TOC) alone, I get a sense of the "story" of the article (check the TOC in the preview after raising the PR or by starting the blog locally on your machine)
- [ ] if using sub-headlines, there's at least two sub-headlines below a main-headline
- [ ] headlines are consistent throughout the article (same level of abstraction, same style - if one headline starts with a verb in imperative form, the other headlines on the same level should probably, too)

## Code Examples
- [ ] code examples are introduced with a ":" and explained below the code example
- [ ] code examples are formatted to avoid horizontal scrolling (manual line breaks and 2-spaces indentation)
- [ ] the text has been checked with [Grammarly](https://grammarly.com) (the free tier is enough)

**Feel free to raise the PR even if the checklist hasn't been worked through, yet**, to take advantage of the preview that is generated for each pull request (see "deploy/netlify" under "Show all checks"). But please only flag it to be reviewed after going through the checklist above.

### CODE REVIEW

File: _posts/2021-05-07-spring-cloud-aws-rds.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+... : Starting SpringcloudrdsApplicationTests using Java 14.0.1 
+...
+...
+Loading class `com.mysql.jdbc.Driver'. This is deprecated. \
+The new driver class is `com.mysql.cj.jdbc.Driver'...
+currentDate 2021-05-12
+... : Shutting down ExecutorService 'applicationTaskExecutor'
+
+```
+We can see a warning in the log for using a deprecated driver class which is safe to be ignored. We have not specified any driver class here. The driver class `com.mysql.jdbc.Driver` is registered based on the metadata read from the database connection to AWS RDS.
```
Comment (thombergs) [inline _posts/2021-05-07-spring-cloud-aws-rds.md:260]: Is there a ticket somewhere in the Spring Cloud AWS project that tracks this deprecation?

File: _posts/2021-05-07-spring-cloud-aws-rds.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+      }
+      
+    });
+    return result;     
+  }
+
+}
+
+```
+Here we have decorated the method `getUsers` with `Transactional(readOnly = true)`. At runtime, all the invocations of this method will be sent to the read-replica.
```
Comment (thombergs) [inline _posts/2021-05-07-spring-cloud-aws-rds.md]: But where did we define the read replica? There is only one data source in the configuration.

File: _posts/2021-05-07-spring-cloud-aws-rds.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+This replication is synchronous as compared to the read-replica described in the previous section.
+
+Spring Cloud AWS JDBC module supports the Multi-AZ failover with a retry interceptor which can be associated with a method to retry any failed transactions during a Multi-AZ failover. The configuration of our retry interceptor is shown below: 
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" ...>
+
+  <jdbc:retry-interceptor
```
Comment (thombergs) [inline _posts/2021-05-07-spring-cloud-aws-rds.md:335]: Do we really have to do this configuration in XML?

**Review by thombergs — CHANGES_REQUESTED** [review]
Nice! I left some nit-picking comments, as usual :)

File: _posts/2021-05-07-spring-cloud-aws-rds.md (hunk 1 above)
Comment (pratikdas) [reply _posts/2021-05-07-spring-cloud-aws-rds.md:260]: It is deprecated by MySQL connector/J api. Spring cloud aws jdbc is still referring to the earlier name (checked the source code)

The name of the class that implements java.sql.Driver in MySQL Connector/J has changed from com.mysql.jdbc.Driver to com.mysql.cj.jdbc.Driver. The old class name has been deprecated.

**Review by thombergs — APPROVED** [review]

**Pull request merged**